lib/SILOptimizer/Transforms/SILMem2Reg.cpp

```diff
@@ -105,9 +106,10 @@ class LiveValues {
       if (!lexicalLifetimeEnsured(asi)) {
         return stored;
       }
-      // We should have created a move of the @owned stored value.
-      assert(move);
-      return move;
+      assert(move || stored->getOwnershipKind() == OwnershipKind::None);
+      // If we had a trivial value stored in a lexical non trivial enum, we may
```

Comment:
Should we have add a third kind `struct None` in addition to `struct Owned` and `struct Guaranteed`?